aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/AuthComponentTest.java

```diff
@@ -0,0 +1,459 @@
+     * @throws JSONException has to be declared as part of creating a test JSON object
+     */
+    @Test
+    public void testConfigure() throws AmplifyException, JSONException {
+        UserStateDetails userStateDetails = new UserStateDetails(UserState.SIGNED_OUT, null);
+        Context context = getApplicationContext();
+        JSONObject testConfig = new JSONObject().put("TestKey", "TestVal");
+
+        doAnswer(invocation -> {
+            assertEquals(context, invocation.getArgument(0));
+            assertEquals(testConfig.toString(), invocation.getArgument(1).toString());
+
+            Callback<UserStateDetails> callback = invocation.getArgument(2);
+            callback.onResult(userStateDetails);
+            return null;
+        }).when(mobileClient).initialize(any(Context.class), any(AWSConfiguration.class), any(Callback.class));
+
+        authPlugin.configure(testConfig, context);
+        verify(mobileClient, times(1))
+                .initialize(any(Context.class), any(AWSConfiguration.class), any(Callback.class));
+    }
+
+    /**
+     * Tests that if AWSMobileClient returns an error callback during initialization, the Auth configure method
+     * throws an AuthException.
+     * @throws AuthException the AuthException expected to be thrown if initialization fails.
+     */
+    @Test(expected = AuthException.class)
+    public void testConfigureExceptionHandling() throws AuthException {
+        doAnswer(invocation -> {
+            Callback<UserStateDetails> callback = invocation.getArgument(2);
+            callback.onError(new Exception());
+            return null;
+        }).when(mobileClient).initialize(any(Context.class), any(AWSConfiguration.class), any(Callback.class));
+
+        authPlugin.configure(new JSONObject(), getApplicationContext());
+    }
+
+    /**
+     * Tests that the signUp method of the Auth wrapper of AWSMobileClient (AMC) calls AMC.signUp() with the username
+     * and password it received and, if included, the userAttributes and validationData sent in the options object.
+     * Also ensures that in the onResult case, the success callback receives a valid AuthSignUpResult.
+     */
+    @Test
+    public void signUp() {
+        SignUpResult amcResult = new SignUpResult(
+            false,
+            new UserCodeDeliveryDetails(
+                    destination,
+                    deliveryMedium,
+                    attributeName
+            ),
+            null
+        );
+
+        doAnswer(invocation -> {
+            assertEquals(username, invocation.getArgument(0));
+            assertEquals(password, invocation.getArgument(1));
+            Map<String, String> attributeMap = invocation.getArgument(2);
+            assertTrue(attributeMap.containsKey(attributeKey));
+            assertEquals(attributeVal, attributeMap.get(attributeKey));
+            assertEquals(metadata, invocation.getArgument(3));
```

Comment:
These are basically just testing your test. The `doAnswer` for faking a behavior on a a component. You'd want to make assertions on the final state, after your test. But here, it's sort of intermediate, before you do `callback.onResult(amcResult)`.

Outside of this `doAnswer().when()` constrcut, later, I'd expect some like `assertTrue(hasSomePropertyOfInterest(amcResult))`.

Reply:
No so what's happening here is I'm testing that AWSMobileClient is being called with the right parameters. The whole point of these tests, per my comments on them, is to ensure that the parameters the user is passing in are then being properly routed to the AWSMobileClient method we're wrapping and that the result which comes back from that method is properly wrapped / converted into the right Amplify result object.